compiler/src/iree/compiler/Codegen/LLVMCPU/test/vectorize_and_hoist.mlir

```diff
@@ -0,0 +1,74 @@
+// RUN: iree-opt --iree-codegen-generic-vectorization=enable-vector-masking\
+// RUN:          --iree-codegen-hoist-redundant-vector-transfers %s | FileCheck --implicit-check-not=mask %s
```

Comment:
I think having this test is fine on its own, but for this pattern change, could we have a simpler one that shows the pattern is not applied with two users?

Reply:
Added: fold_extract_into_transfer_read.mlir